server/shadow/shadow_server.c

```diff
@@ -510,10 +519,12 @@ int shadow_server_start(rdpShadowServer* server)
 		return -1;
 	}
 
-	if (!server->ipcSocket)
-		status = server->listener->Open(server->listener, NULL, (UINT16)server->port);
-	else
-		status = server->listener->OpenLocal(server->listener, server->ipcSocket);
+	// Bind to TCP port if nothing is specified, otherwise bind to everything specified
+	if (!server->ipcSocket || server->bindAddress)
+		status &=
+		    server->listener->Open(server->listener, server->bindAddress, (UINT16)server->port);
```

Comment:
Sure you want to change the `if(){ else {}` this way? the whole change is not necessary, `server->bindAddress` should simply be initialized as `NULL`

Reply:
I think so—since we want to bind both sockets if both options are given, and default to binding just TCP if neither of the options are given.